### PR TITLE
Upgrade to the latest emscripten

### PR DESCRIPTION
**Pull request by dfellis**
The latest emscripten also supports compiling to WASM and using WebAssembly instead.

Just having WASM turned on caused issues, but upgrading the compiler building the artifact is the first step in adding WebAssembly support for faster browser performance that can pair nicely with the faster performance in [h3-node](https://github.com/dfellis/h3-node) :)

(Also I updated my email address since that one doesn't actually work, anymore.)

An alternative may be to fork `h3-js` to do a WASM-only `h3-wasm`, but this particular compiler upgrade still seems like a good idea at least for bugfixes and preventing bitrot.

@nrabinowitz Which would you prefer? (Also, perhaps we should ask the owner of [this package](https://www.npmjs.com/package/h3) if they'd be willing to give us the name and that can be a meta package that loads `h3-js`/`h3-node`/`h3-wasm` depending on the JS environment capabilities? :smile: )

### CODE REVIEW

File: .build-emscripten.sh [hunk 1]
```diff
@@ -36,8 +36,9 @@ pushd src/h3lib/lib
 cp ../../../../build/sizes.h ../include
 cp ../../../../build/sizes.c .
 # Compile with emscripten
-emcc -O3 -I ../include *.c -o libh3.js -DH3_HAVE_VLA -s INVOKE_RUN=0 -s EXPORT_NAME="'libh3'" -s MODULARIZE=1 -s NO_FILESYSTEM=1 -s NODEJS_CATCH_EXIT=0 -s TOTAL_MEMORY=33554432 -s ALLOW_MEMORY_GROWTH=1 -s WARN_UNALIGNED=1 -s EXPORTED_FUNCTIONS=$bound_functions -s EXTRA_EXPORTED_RUNTIME_METHODS='["cwrap", "getValue", "setValue"]' --memory-init-file 0
+emcc -O3 -I ../include *.c -o libh3.js -DH3_HAVE_VLA -s WASM=0 -s INVOKE_RUN=0 -s EXPORT_NAME="'libh3'" -s MODULARIZE=1 -s NO_FILESYSTEM=1 -s NODEJS_CATCH_EXIT=0 -s TOTAL_MEMORY=33554432 -s ALLOW_MEMORY_GROWTH=1 -s WARN_UNALIGNED=1 -s EXPORTED_FUNCTIONS=$bound_functions -s EXTRA_EXPORTED_RUNTIME_METHODS='["cwrap", "getValue", "setValue"]' --memory-init-file 0
 cp libh3.js ../../../../out/libh3.js
 echo "module.exports = libh3();" >> ../../../../out/libh3.js
+echo "Object.keys(module.exports.asmLibraryArg).filter(function(k) { return typeof module.exports.asmLibraryArg[k] === 'function'; }).forEach(function(k) { module.exports[module.exports.asmLibraryArg[k].name] = module.exports.asmLibraryArg[k]; });" >> ../../../../out/libh3.js
```
Comment (nrabinowitz) [inline .build-emscripten.sh]: For my info - why is this necessary now? Can you please add a comment explaining what's going on?

This might be cleaner assigning `module.exports` at the end, using arrow functions, and doing only one file append:
```
echo "const h3 = libh3(); Object.keys(h3.asmLibraryArg).filter(k => typeof h3.asmLibraryArg[k] === 'function').forEach(k => { h3[h3.asmLibraryArg[k].name] = h3.asmLibraryArg[k]; });" >> ../../../../out/libh3.js
```
Might be even nicer with heredoc:
```
cat << EOF >> ../../../../out/libh3.js
const h3 = libh3();
Object.keys(h3.asmLibraryArg)
  .filter(k => typeof h3.asmLibraryArg[k] === 'function')
  .forEach(k => {
    h3[h3.asmLibraryArg[k].name] = h3.asmLibraryArg[k];
  });
EOF
```

Comment (dfellis) [reply .build-emscripten.sh]: So the latest version of Emscripten assumes that you're using WASM and not Asm.js and has dropped access to these helper functions from the Asm.js side of things (accidentally, I believe). I was planning to register an issue with Emscripten about this problem and reference this PR.

I found the methods still existing for use by the Asm.js code itself, but not for code interfacing with the Asm.js code and this reattaches these methods as they were before.

Your suggestions on how to format the code in the script are definitely good ones that I'll incorporate, though. :)

Comment (nrabinowitz) [reply .build-emscripten.sh]: Looks good, please add an explanatory comment here and then I think we're good to merge.

File: .build-emscripten.sh [hunk 2]
```diff
@@ -36,8 +36,17 @@ pushd src/h3lib/lib
...
 cp ../../../../build/sizes.c .
 # Compile with emscripten
-emcc -O3 -I ../include *.c -o libh3.js -DH3_HAVE_VLA -s INVOKE_RUN=0 -s EXPORT_NAME="'libh3'" -s MODULARIZE=1 -s NO_FILESYSTEM=1 -s NODEJS_CATCH_EXIT=0 -s TOTAL_MEMORY=33554432 -s ALLOW_MEMORY_GROWTH=1 -s WARN_UNALIGNED=1 -s EXPORTED_FUNCTIONS=$bound_functions -s EXTRA_EXPORTED_RUNTIME_METHODS='["cwrap", "getValue", "setValue"]' --memory-init-file 0
+emcc -O3 -I ../include *.c -o libh3.js -DH3_HAVE_VLA -s WASM=0 -s INVOKE_RUN=0 -s EXPORT_NAME="'libh3'" -s MODULARIZE=1 -s NO_FILESYSTEM=1 -s NODEJS_CATCH_EXIT=0 -s TOTAL_MEMORY=33554432 -s ALLOW_MEMORY_GROWTH=1 -s WARN_UNALIGNED=1 -s EXPORTED_FUNCTIONS=$bound_functions -s EXTRA_EXPORTED_RUNTIME_METHODS='["cwrap", "getValue", "setValue"]' --memory-init-file 0
 cp libh3.js ../../../../out/libh3.js
-echo "module.exports = libh3();" >> ../../../../out/libh3.js
+cat << EOF >> ../../../../out/libh3.js
+const h3 = libh3();
+module.exports = h3;
+// Regression in latest Emscripten drops these methods, re-attach to the object with proper names
```
Comment (nrabinowitz) [inline .build-emscripten.sh]: Something in the build objects to this comment - maybe just use a `#` comment in the script instead?

Comment (dfellis) [reply .build-emscripten.sh]: Very strange, it built and ran tests locally just fine for me. Let me investigate a bit more, I'd prefer to have that comment in the output file as well for that extra level of documentation.

Comment (dfellis) [reply .build-emscripten.sh]: Ahh, I see. I didn't commit the `out/libh3.js` file in that last commit and it aborted because of the difference between the two, should be quick to fix.

Comment (nrabinowitz) [reply .build-emscripten.sh]: Yes, sorry, just realized that as well. That should fix the build.
